src/pipeline.py

```diff
@@ -1,12 +1,21 @@
 """Pipeline used to create a stable diffusion dataset from a set of initial prompts."""
 from pathlib import Path
-
+import torch
 import pyarrow as pa
-from fondant.pipeline import Pipeline
+from fondant.pipeline import Pipeline, Resources
 
 from components.generate_prompts import GeneratePromptsComponent
 
 
+def get_local_resources():
+    if torch.cuda.is_available():
+        return Resources(
+            accelerator_name="GPU", accelerator_number=torch.cuda.device_count()
+        )
+    else:
+        return None
```

Comment:
Does this need to be dynamic?

What happens if we set it to GPU without a GPU being available?